framework/include/cppmicroservices/AnyMap.h

```diff
@@ -221,8 +223,11 @@ class US_Framework_EXPORT any_map
 
   any_map(map_type type);
```

Comment:
Some additional comments to explain some of the background and reasoning for the decisions that were made to improve efficiency would be helpful.